### ISSUE TITLE
[BUG] Cant build with Carthage

### ISSUE THREAD

**Issue by sebastianvarela** [Adyen/adyen-ios#631]
**Describe the bug**
Cant build the last version with Carthage

**To Reproduce**
1. Add the dependency on Cartfile
2. run `carthage update --platform iOS`

**Expected behavior**
Compile all frameworks

**Screenshots**
<img width="1319" alt="Captura de pantalla 2021-09-21 a las 13 03 49" src="https://user-images.githubusercontent.com/2905467/134159778-10d9c46a-ff97-4837-8974-f337065a04be.png">

We test in two Macs:
MacBook Pro Intel, Big Sur, Xcode 12.5.1, Carthage 0.38.0

**Comment by mohammedDehairy** [comment]
Hi @sebastianvarela ,

There is warning in the readMe file about `AdyenWeChatPay` Module:
![Screen Shot 2021-09-21 at 1 18 16 PM](https://user-images.githubusercontent.com/2619933/134161592-c93dc5ef-ebd4-46b3-aa00-da00e05a9a5d.png)

So you need to exclude arm64 simulator architecture if you want to include `AdyenWeChatPay` module, using this script work around:

https://github.com/Carthage/Carthage/blob/master/Documentation/Xcode12Workaround.md

and if you don't want `AdyenWeChatPay` Module, then you can ignore the error and use the other modules as normal.


`AdyenWeChatPay` Module depends on WeChat iOS SDK which is a static .a library that just doesn't support arm64 simulator architecture unfortunately .

Mohamed

**Comment by descorp** [comment]
Hey @sebastianvarela 

Was the issue resolved?

**Comment by sebastianvarela** [comment]
> Hey @sebastianvarela
> 
> Was the issue resolved?

no, I can't compile with Carthage with this repo, I had to make a fork to compile without WeChatPay Scheme by now. 

I the near future we test again with Xcode 13.

**Comment by descorp** [comment]
Let's verify:

1) On Xcode 12 you have to use [workaround](https://github.com/Carthage/Carthage/blob/master/Documentation/Xcode12Workaround.md)
2) We expect `AdyenWeChatPay` to fail on Carthages - even thought the `carthage_workaround update --platform iOS` will fail, other modules will be build as expected and you should be able to add 'Adyen', 'AdyenDropIn' etc to your project as usual.

**Comment by descorp** [comment]
Hey @sebastianvarela 

how is you status? Do you want to proceed investigation?

**Comment by descorp** [comment]
Hey @r-pedraza 

If you are running Carthage on **Intel Mac** via Xcode 13 using:

[carthage.sh](https://github.com/Adyen/adyen-ios/blob/develop/Scripts/carthage.sh) update --use-xcframeworks

Your output will look somewhat like this:

```bash
*** Fetching ．．
*** Fetching adyen-networking-ios
*** Fetching adyen-3ds2-ios
*** Checking out ．． at "400f57d4f46db84fb66cafbd0768f5520cad9fc5"
*** Checking out adyen-3ds2-ios at "2.2.3"
*** Checking out adyen-networking-ios at "1.0.0"
*** xcodebuild output can be found in /var/folders/wg/2190p3pn00s29ptng0dq_wc00000z9/T/carthage-xcodebuild.4yLMmk.log
*** Downloading adyen-3ds2-ios binary at "2.2.3"
*** Building scheme "AdyenNetworking" in AdyenNetworking.xcodeproj
*** Building scheme "Adyen" in Adyen.xcodeproj
*** Building scheme "AdyenCard" in Adyen.xcodeproj
*** Building scheme "AdyenComponents" in Adyen.xcodeproj
*** Building scheme "AdyenActions" in Adyen.xcodeproj
*** Building scheme "AdyenEncryption" in Adyen.xcodeproj
*** Building scheme "AdyenDropIn" in Adyen.xcodeproj
*** Building scheme "AdyenSwiftUI" in Adyen.xcodeproj
*** Building scheme "AdyenWeChatPay" in Adyen.xcodeproj
Build Failed
	Task failed with exit code 65:
	....
```

As mentioned above - failure of `AdyenWeChatPay` **is expected** (embedded WeChat binaries are not compatible with `arm64`), however this failure **does not effect other targets**.

You should be able to find a perfectly build `.xcframeworks` in `Carthage/Build`:

![Screenshot 2021-10-19 at 23 41 40](https://user-images.githubusercontent.com/2648655/137994846-623b0f6f-a810-49e9-b18b-c13890a4d7c0.png)

**Comment by descorp** [comment]
In v4.5.0 we have updated Carthage script. It should provide better compatibility with Xcode 13.

Due to the lack of activity, I am going to close this ticket.
Feel free to reopen if you need additional assistance.

**Comment by sebastianvarela** [comment]
I dont see how to reopen this issue, you can reopen?? please.

**Comment by descorp** [comment]
Hey @sebastianvarela 

We are facing similar issue on latest Carthage 0.38 locally, but, somehow, it successfully build on [our CI](https://github.com/Adyen/adyen-ios/runs/4919452243?check_suite_focus=true) and few dev machines.

Could you check the content of `xcodebuild log`?
Can you see anything that look like en error / reason for failure?

**Comment by sebastianvarela** [comment]
@descorp : I tried, but fails:

```
*** Fetching adyen-ios
*** Fetching adyen-networking-ios
*** Fetching adyen-3ds2-ios
*** Checking out adyen-networking-ios at "1.0.0"
*** Checking out adyen-ios at "07f9d3a965e10669291e5fb06784f3e0fdc27b53"
*** Checking out adyen-3ds2-ios at "2.2.4"
*** xcodebuild output can be found in /var/folders/_6/80g1b_n50ls3k4j29dlgj7gw0000gn/T/carthage-xcodebuild.DyDqBl.log
*** Valid cache found for adyen-3ds2-ios, skipping build
*** Valid cache found for adyen-networking-ios, skipping build
*** Invalid cache found for adyen-ios, rebuilding with all downstream dependencies
*** Building scheme "AdyenComponents" in Adyen.xcodeproj
*** Building scheme "AdyenCard" in Adyen.xcodeproj
*** Building scheme "AdyenActions" in Adyen.xcodeproj
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -project /Users/sebastian.varela/repos/masmovil/ragnarok-ios/Carthage/Checkouts/adyen-ios/Adyen.xcodeproj -scheme AdyenActions -configuration Release -derivedDataPath /Users/sebastian.varela/Library/Caches/org.carthage.CarthageKit/DerivedData/13.2.1_13C100/adyen-ios/07f9d3a965e10669291e5fb06784f3e0fdc27b53 -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive VALIDATE_WORKSPACE=NO -archivePath /var/folders/_6/80g1b_n50ls3k4j29dlgj7gw0000gn/T/adyen-ios SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO FRAMEWORK_SEARCH_PATHS=$(inherited)\ /var/folders/_6/80g1b_n50ls3k4j29dlgj7gw0000gn/T/carthage-xcframework-qK2G (launched in /Users/sebastian.varela/repos/masmovil/ragnarok-ios/Carthage/Checkouts/adyen-ios)
```

LOG:
[carthage-xcodebuild.DyDqBl.log.zip](https://github.com/Adyen/adyen-ios/files/7958734/carthage-xcodebuild.DyDqBl.log.zip)

**Comment by descorp** [comment]
Hm.. `no such module 'Adyen3DS2'`
If you run it again - do you have same error?

```
*** Valid cache found for adyen-3ds2-ios, skipping build
*** Valid cache found for adyen-networking-ios, skipping build
*** Invalid cache found for adyen-ios, rebuilding with all downstream dependencies
```

Could you clean up cache (delete `Carthage` folder)?

**Comment by sebastianvarela** [comment]
@descorp : I clean Carthage cache and folder:
```bash
rm -rf ~/Library/Caches/org.carthage.CarthageKi
```

and later:
```bash
*** Checking out adyen-networking-ios at "1.0.0"
*** Checking out adyen-ios at "07f9d3a965e10669291e5fb06784f3e0fdc27b53"
*** Checking out adyen-3ds2-ios at "2.2.4"
*** No cache found for adyen-3ds2-ios, building with all downstream dependencies
*** No cache found for adyen-networking-ios, building with all downstream dependencies
*** xcodebuild output can be found in /var/folders/_6/80g1b_n50ls3k4j29dlgj7gw0000gn/T/carthage-xcodebuild.YyknOO.log
*** Skipped building adyen-3ds2-ios due to the error:
Dependency "adyen-3ds2-ios" has no shared framework schemes for any of the platforms: iOS

If you believe this to be an error, please file an issue with the maintainers at https://github.com/adyen/adyen-3ds2-ios/issues/new
*** Building scheme "AdyenNetworking" in AdyenNetworking.xcodeproj
*** Building scheme "AdyenComponents" in Adyen.xcodeproj
*** Building scheme "AdyenCard" in Adyen.xcodeproj
*** Building scheme "AdyenActions" in Adyen.xcodeproj
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -project /Users/sebastian.varela/repos/masmovil/ragnarok-ios/Carthage/Checkouts/adyen-ios/Adyen.xcodeproj -scheme AdyenActions -configuration Release -derivedDataPath /Users/sebastian.varela/Library/Caches/org.carthage.CarthageKit/DerivedData/13.2.1_13C100/adyen-ios/07f9d3a965e10669291e5fb06784f3e0fdc27b53 -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive VALIDATE_WORKSPACE=NO -archivePath /var/folders/_6/80g1b_n50ls3k4j29dlgj7gw0000gn/T/adyen-ios SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO FRAMEWORK_SEARCH_PATHS=$(inherited)\ /var/folders/_6/80g1b_n50ls3k4j29dlgj7gw0000gn/T/carthage-xcframework-yecC (launched in /Users/sebastian.varela/repos/masmovil/ragnarok-ios/Carthage/Checkouts/adyen-ios)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/_6/80g1b_n50ls3k4j29dlgj7gw0000gn/T/carthage-xcodebuild.YyknOO.log
```

The log said:
```bash
/Users/sebastian.varela/repos/masmovil/ragnarok-ios/Carthage/Checkouts/adyen-ios/AdyenActions/Components/3DS2/ThreeDS2ComponentChallengeToken.swift:7:8: error: no such module 'Adyen3DS2'
import Adyen3DS2
       ^

** ARCHIVE FAILED **


The following build commands failed:
	CompileSwift normal arm64 (in target 'AdyenActions' from project 'Adyen')
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler (in target 'AdyenActions' from project 'Adyen')
(2 failures)
```

My Cartfile/Cartfile.resolved:
```bash
github "adyen/adyen-ios" "m1_fix"
```
```bash
github "adyen/adyen-3ds2-ios" "2.2.4"
github "adyen/adyen-ios" "07f9d3a965e10669291e5fb06784f3e0fdc27b53"
github "adyen/adyen-networking-ios" "1.0.0"
```

**Comment by descorp** [comment]
hm..

this is really odd:

```
*** Skipped building adyen-3ds2-ios due to the error:
Dependency "adyen-3ds2-ios" has no shared framework schemes for any of the platforms: iOS
```

Can you just try `carthage update` with only `github "adyen/adyen-3ds2-ios" "2.2.4"` ?
I haven't changed since [November](https://github.com/Adyen/adyen-3ds2-ios/releases/tag/2.2.4).

**Comment by sebastianvarela** [comment]
@descorp : it's because of the `--no-use-binaries` flag
But that is not a problem, since what is failing is something else (I take the binary from checkout)

**Comment by descorp** [comment]
@sebastianvarela 

>  what is failing is something else

In logs that you have shared previously I see that `Adyen3DS2` is missing.

**Comment by descorp** [comment]
We are working on it :)

Could you verify that it works for you and you can build for Simulator and Device?

**Comment by descorp** [comment]
Hey @sebastianvarela 

We have merged `m1_fix` branch.
Could you check if it fixes Carthage for you?

**Comment by descorp** [comment]
Fixed in [4.7.0](https://github.com/Adyen/adyen-ios/releases/tag/4.7.0)

@sebastianvarela could you verify?

**Comment by sebastianvarela** [comment]
Done! thanks!

```bash
# sh bin/carthage_workaround.sh bootstrap --platform iOS --cache-builds --use-xcframeworks
*** Checking out adyen-ios at "4.7.0"
*** Checking out adyen-networking-ios at "1.0.0"
*** Checking out adyen-3ds2-ios at "2.2.4"
*** Checking out adyen-wechatpay-ios at "2.1.0"
*** No cache found for adyen-3ds2-ios, building with all downstream dependencies
*** No cache found for adyen-networking-ios, building with all downstream dependencies
*** No cache found for adyen-wechatpay-ios, building with all downstream dependencies
*** xcodebuild output can be found in /var/folders/_6/80g1b_n50ls3k4j29dlgj7gw0000gn/T/carthage-xcodebuild.vDQWrF.log
*** Downloading adyen-3ds2-ios binary at "2.2.4"
*** Downloading adyen-wechatpay-ios binary at "2.1.0"
*** Building scheme "AdyenNetworking" in AdyenNetworking.xcodeproj
*** Building scheme "AdyenCard" in Adyen.xcodeproj
*** Building scheme "AdyenComponents" in Adyen.xcodeproj
*** Building scheme "AdyenActions" in Adyen.xcodeproj
*** Building scheme "Adyen" in Adyen.xcodeproj
*** Building scheme "AdyenDropIn" in Adyen.xcodeproj
*** Building scheme "AdyenEncryption" in Adyen.xcodeproj
*** Building scheme "AdyenSwiftUI" in Adyen.xcodeproj
*** Building scheme "AdyenWeChatPay" in Adyen.xcodeproj
```
